pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/predicate/RegexPredicateEvaluator.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.operator.filter.predicate;
+
+import java.util.regex.Pattern;
+
+import com.linkedin.pinot.core.common.predicate.EqPredicate;
+import com.linkedin.pinot.core.common.predicate.RegexPredicate;
+import com.linkedin.pinot.core.segment.index.readers.Dictionary;
+
+public class RegexPredicateEvaluator implements PredicateEvaluator {
+
+  private RegexPredicate predicate;
+  private Dictionary dictionary;
+  private Pattern pattern;
+
+  public RegexPredicateEvaluator(RegexPredicate predicate, Dictionary dictionary) {
```

Comment:
Just a heads up, there's another PR in-flight which requires no-dictionary support for predicate evaluators.